src/github.com/getlantern/errlog/errlog.go

```diff
@@ -0,0 +1,529 @@
+/*
+Package errlog defines error types used across Lantern project and implements
+functions to manipulate them.
+
+  var elog = errlog.ErrorLoggerFor("package-name")
+  if n, err := Foo(); err != nil {
+    elog.Log(err, errlog.WithOp("foo"), errlog.WithField("foo": "bar"))
+  }
+
```

Comment:
An alternative is free style fields similar to https://github.com/go-kit/kit/tree/master/log. As the `errlog` package is used only by us at this point, I chose more rigid syntax.